packages/runtime/test-runtime-utils/src/test/insecureUrlResolverTest.spec.ts

```diff
@@ -48,6 +49,29 @@ describe("Insecure Url Resolver Test", () => {
         const response = await resolver.getAbsoluteUrl(resolvedUrl, dataStoreId);
 
         const compUrl = `${hostUrl}/${tenantId}/${fileName}/${dataStoreId}`;
-        assert.equal(response, compUrl, "Url should match");
+        assert.strictEqual(response, compUrl, "Url should match");
+    });
+
+    it("Test RequestUrl for longer url", async () => {
+        const testRequest: IRequest = {
+            url: `https://localhost/${fileName}/random/random2`,
+            headers: {},
+        };
+
+        if (typeof window === "undefined" && typeof global === "object") {
```

Comment:
Add a comment here that we're mocking the window since InsecureUrlResolver depends on window.location.